src/preproc/preprocessing_pass_registry.cpp

```diff
@@ -0,0 +1,56 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_registry.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Preprocessing pass registry
+ **
+ ** Preprocessing pass registry that uses a name to store instances of
+ ** classes in a map. Makes sure passes are properly disposed of when
+ ** registry is destroyed.  
+ **/
+#include "preproc/preprocessing_pass_registry.h"
+#include <utility>
+#include <cassert>
+#include "base/output.h"
+#include "preproc/preprocessing_pass.h"
+
+namespace CVC4 {
+namespace preproc {
+
+void PreprocessingPassRegistry::init(PreprocessingPassAPI* api) {
+  for (std::pair<std::string, PreprocessingPass*>&& element :
+       d_stringToPreprocessingPass) {
+    element.second->init(api);
+  }
+}
+
+void PreprocessingPassRegistry::registerPass(
+    const std::string& ppName, PreprocessingPass* preprocessingPass) {
+  Debug("pp-registry") << "Registering pass " << ppName << std::endl;
+  Assert(d_stringToPreprocessingPass[ppName] == NULL);
+  d_stringToPreprocessingPass[ppName] = preprocessingPass;
+}
+
+PreprocessingPass* PreprocessingPassRegistry::getPass(
+    const std::string& ppName) {
+  assert(d_stringToPreprocessingPass.find(ppName) !=
```

Comment:
Sorry, in my previous comment, I suggested to include `<cassert>` but it would actually be better to change this to an `Assert` (the CVC4 version of an assert) and to include `base/cvc4_assert.h` instead of `cassert`.